pkg/rebuild/rebuild/rebuildremote.go

```diff
@@ -110,39 +127,71 @@ var assetUploadTpl = template.Must(
 	))
 
 func makeBuild(t Target, dockerfile string, opts RemoteOptions) (*cloudbuild.Build, error) {
+	var finalizeScript bytes.Buffer
+	{
+		err := rebuildFinalizeTpl.Execute(&finalizeScript, map[string]any{
+			"Artifact": transfer{
+				From: path.Join("/out", t.Artifact),
+				To:   path.Join("/workspace", t.Artifact),
+			},
+			"EnableEBPF": opts.EnableEBPF,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "expanding asset upload template")
+		}
+	}
 	var assetUploadScript bytes.Buffer
-	err := assetUploadTpl.Execute(&assetUploadScript, map[string]any{
-		"UtilPrebuildBucket": opts.UtilPrebuildBucket,
-		"Uploads": []upload{
+	{
+		var uploads = []transfer{
 			{From: "/workspace/image.tgz", To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: ContainerImageAsset}).String()},
 			{From: path.Join("/workspace", t.Artifact), To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: RebuildAsset}).String()},
-		},
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "expanding asset upload template")
+		}
+		if opts.EnableEBPF {
+			uploads = append(uploads, transfer{From: "/workspace/tetragon.jsonl", To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: TetragonLog}).String()})
+		}
+		err := assetUploadTpl.Execute(&assetUploadScript, map[string]any{
+			"UtilPrebuildBucket": opts.UtilPrebuildBucket,
+			"Uploads":            uploads,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "expanding asset upload template")
+		}
 	}
+	steps := []*cloudbuild.BuildStep{}
+	if opts.EnableEBPF {
+		steps = append(steps, &cloudbuild.BuildStep{
+			Name:         "gcr.io/cloud-builders/docker",
+			Args:         []string{"run", "--name=tetragon", "--pid=host", "--cgroupns=host", "--privileged", "-v=/sys/kernel/btf/vmlinux:/var/lib/tetragon/btf", "quay.io/cilium/tetragon:v1.1.2", "/usr/bin/tetragon", "--export-filename=/var/log/tetragon/tetragon.jsonl"},
+			Id:           "run_tetragon",
+			AllowFailure: true,
+		})
+	}
+	steps = append(steps, []*cloudbuild.BuildStep{
+		{
+			Name:   "gcr.io/cloud-builders/docker",
+			Script: "set -eux\ncat <<'EOS' | docker buildx build --tag=img -\n" + dockerfile + "\nEOS\ndocker run --name=container img",
+			Id:     "run_builder",
+			// Run immediately ("-" is a special value indicating this step
+			// shouldn't wait for anything).
+			WaitFor: []string{"-"},
+		},
+		{
+			Name:    "gcr.io/cloud-builders/docker",
+			Script:  finalizeScript.String(),
+			Id:      "finalize",
+			WaitFor: []string{"run_builder"},
```

Comment:
When WaitFor is omitted, the steps will wait for all previous steps to complete. https://cloud.google.com/build/docs/build-config-file-schema#waitfor

Because we're expecting the tetragon build step to still be running, that default behavior doesn't work. If you wanted to reduce the number of bytes we could just use numbers for the step ids.  WDYT?

Reply:
gotcha. per offline conversation, an alternative here would be to avoid the GCB config and just inline the syscall monitor into our two current build templates. This won't necessarily scale if we add more instrumentation but given the simplicity of this one-liner and us not having imminent plans to add more observability components, I think using the templating approach probably gets us the most bang for our buck with the least amount of added complexity.